test/unit/mdc-textfield/foundation.test.js

```diff
@@ -105,6 +107,19 @@ test('#init adds mdc-textfield--upgraded class', (t) => {
   t.end();
 });
 
+test('#setKeyInteraction sets keyInteractionOccurred to true', (t) => {
```

Comment:
We try not to test private methods, besides simply referencing `adapter_` in component tests. The reason is bc private methods are considered implementation details; a means to an end if you will. And what we really want is to be testing the ends. Specifically:

- Focused styles are applied when an `input` event is triggered on an element and no keydown was triggered beforehand (or, per the comment above,  some other means, such as if the element is not focused).

So it looks like that's just one single event test that should be implemented.